tfhe/src/typed_api/traits.rs

```diff
@@ -0,0 +1,123 @@
+use crate::typed_api::ClientKey;
+
+/// Trait used to have a generic way of creating a value of a FHE type
+/// from a native value.
+///
+/// This trait is for when FHE type the native value is encrypted
+/// supports the same numbers of bits of precision.
+///
+/// The `Key` is required as it contains the key needed to do the
+/// actual encryption.
+pub trait FheEncrypt<T, Key> {
+    fn encrypt(value: T, key: &Key) -> Self;
+}
+
+pub trait DynamicFheEncryptor<T> {
+    type FheType;
+
+    fn encrypt(&self, value: T, key: &ClientKey) -> Self::FheType;
+}
+
+// This trait has the same signature than
+// `std::convert::From` however we create our own trait
+// to be explicit about the `trivial`
```

Comment:
missing a leve of / to have a docstring ?

Reply:
Its not meant to be a docstring, it could be, but it just a comment as to way its done this way

Reply:
ok up to you, feel free to mark as resolved